packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/project/services/ProjectSemanticDataExportParticipant.java

```diff
@@ -72,25 +72,25 @@ public ProjectSemanticDataExportParticipant(IIdentityService identityService, IE
     }
 
     @Override
-    public Map<String, Object> exportData(Project project, ZipOutputStream outputStream) {
+    public Map<String, Object> exportData(Project project, String editingContextId, ZipOutputStream outputStream) {
         Map<String, Object> manifestEntries = new HashMap<>();
 
-        var optionalEditingContext = this.projectSemanticDataSearchService.findByProjectId(AggregateReference.to(project.getId()))
-                .map(ProjectSemanticData::getSemanticData)
-                .map(AggregateReference::getId)
-                .map(UUID::toString)
-                .flatMap(editingContextSearchService::findById)
-                .filter(IEMFEditingContext.class::isInstance)
-                .map(IEMFEditingContext.class::cast);
+        var optionalEditingContext = this.projectSemanticDataSearchService.findBySemanticDataId(AggregateReference.to(UUID.fromString(editingContextId)))
+            .map(ProjectSemanticData::getSemanticData)
```

Comment:
Let's say someone give your the id of a project A and the editingContextId of a project B, they can retrieve data from both sources. This is exactly the kind of way one could use to retrieve data from a project they do not have access to.

Here you are also considering that you know that the editing context id is a UUID, which is not always the case and you are parsing it without the appropriate class. I specifically asked for `name` as a parameter and for the backend to look by the project semantic data name in order to ensure that we couldn't export inconsistent data.

Please follow the pieces of advice that you ask me for.